halo2_gadgets/benches/poseidon.rs

```diff
@@ -221,30 +169,40 @@ fn bench_poseidon<S, const WIDTH: usize, const RATE: usize, const L: usize>(
 
     let circuit = HashCircuit::<S, WIDTH, RATE, L> {
         message: Value::known(message),
-        output: Value::known(output),
         _spec: PhantomData,
     };
 
+    // Create a proof
+    let mut transcript = Blake2bWrite::<_, _, Challenge255<_>>::init(vec![]);
+
     c.bench_function(&prover_name, |b| {
         b.iter(|| {
-            // Create a proof
-            let mut transcript = Blake2bWrite::<_, _, Challenge255<_>>::init(vec![]);
```

Comment:
Hmm, this has the side effect that it will continually append to the transcript. Depending on the size of the circuit (small in this case) and the number of iterations, this pattern could lead to an unintentional OOM. It also causes subsequent iterations to have different memory reallocation patterns, though given that allocations double I suspect what it means is that it slightly reduces the benchmarked time.